test/jasmine/tests/cartesian_interact_test.js

```diff
@@ -2547,3 +2547,123 @@ describe('Cartesian plots with css transforms', function() {
         });
     });
 });
+
+describe('Cartesian taces with zindex', function() {
+    var gd;
+
+    beforeEach(function() {
+        gd = createGraphDiv();
+    });
+
+    afterEach(destroyGraphDiv);
+
+    var data0 = [
+        {x: [1, 2], y: [1, 1], type: 'scatter', zindex: 10},
+        {x: [1, 2], y: [1, 2], type: 'scatter'},
+        {x: [1, 2], y: [1, 3], type: 'scatter', zindex: 5}
+    ];
+
+    var data1 = [
+        {x: [1, 2], y: [1, 1], type: 'scatter'},
+        {x: [1, 2], y: [1, 2], type: 'scatter', zindex: -5},
+        {x: [1, 2], y: [1, 3], type: 'scatter', zindex: 10},
+    ];
+
+    function fig(data) {
+        return {
+            data: data,
+            layout: {
+                width: 400, height: 400
+            }
+        };
+    }
+
+    function assertZIndices(data, expectedData) {
+        for(var i = 0; i < data.length; i++) {
+            var zindex = expectedData[i].zindex ? expectedData[i].zindex : 0;
+            expect(data[i].zindex).toEqual(zindex);
+        }
+    }
+
+    function assertZIndicesSorted(data) {
+        var prevZIndex;
+        expect(data.length).toBeGreaterThan(0);
+        for(var i = 0; i < data.length; i++) {
+            var currentZIndex = data[i].__data__.zindex;
+            if(prevZIndex !== undefined) {
+                expect(currentZIndex).toBeGreaterThanOrEqual(prevZIndex);
+            }
+            prevZIndex = currentZIndex;
+        }
+    }
+
+    it('should be able to update zindex', function(done) {
```

Comment:
```suggestion
    it('should be able to update and remove layers for scatter traces in respect to zindex', function(done) {
```

Reply:
I've added a separate test to remove layers. Do you think I should append it to this test?

Reply:
Possibly yes so that we add it for other traces e.g. bar next.